src/org/etools/j1939_84/modules/SupportedSpnModule.java

```diff
@@ -341,4 +366,43 @@ public boolean validateFreezeFrameSpns(ResultsListener listener, Collection<Inte
         return result;
     }
 
+    private static Collection<SpnGroup> getNoticedNoDupDataStreamSpns(FuelType fuelType, int engineModelYear) {
+        Collection<SpnGroup> requiredSpns = new HashSet<>(NOTICED_NON_DUP_DATA_STREAM_SPNS);
+
+        if (engineModelYear >= 2024) {
+            requiredSpns.addAll(NOTICED_NON_DUP_2024_DATA_STREAM_SPNS);
+        }
+
+        return requiredSpns.stream().sorted().collect(Collectors.toList());
+    }
+
+    private static Collection<SpnGroup> getDesiredNoDupDataStreamSpns(FuelType fuelType, int engineModelYear) {
+        Collection<SpnGroup> requiredSpns = new HashSet<>(DESIRED_NON_DUP_DATA_STREAM_SPNS);
+
+        return requiredSpns.stream().sorted().collect(Collectors.toList());
+    }
+
+
+    public boolean isMoreThanOneSpnReportedInfo(ResultsListener listener, Collection<Integer> spns, FuelType fuelType, int engineModelYear){
```

Comment:
These methods don't seem to be used and they are missing unit tests.

Reply:
The code is coming that it is used in.

Reply:
It is used in the part 1 step 4 which is also up for review.  I have tickets in for the unit tests.